### PR TITLE
RivetAnalyzer consumes migration 76X

### PR DESCRIPTION
**Pull request by jhgoh**
This PR is for the consumes() migration of  RivetAnalyzer module in GeneratorInterface/RivetInterface.
It is purely a technical update.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
hold
Do we know that the rivet code is thread safe? That is, does it use statics or other global variables internally?

**Comment by Dr15Jones** [comment]
I removed my last comment since it wasn't the Rivet calling producer which was changed to a stream producer, it was another one. Sorry for the confusion.

File: GeneratorInterface/RivetInterface/python/genParticles2HepMC_cfi.py [hunk 1]
```diff
@@ -2,5 +2,5 @@
 
 generator = cms.EDProducer("GenParticles2HepMCConverter",
     genParticles = cms.InputTag("genParticles"),
-    genEventInfo = cms.InputTag("generator"),
+    genEventInfo = cms.InputTag("generator", "", "SIM"),
```
Comment (davidlange6) [inline GeneratorInterface/RivetInterface/python/genParticles2HepMC_cfi.py]: what is the motivation for this change?

Comment (jhgoh) [inline GeneratorInterface/RivetInterface/python/genParticles2HepMC_cfi.py]: Without specifying the process name, it raises some warning because this module tries to read object with same name:
%MSG-e UnrunnableSchedule:  AfterModConstruction 31-Jul-2015 14:51:13 CEST  pre-events
Module run order problem found: 
```generator consumes generator
 Running in the threaded framework would lead to indeterminate results.
 Please change order of modules in mentioned Path(s) to avoid inconsistent module ordering.
%MSG```
Of course one can change the name of GenParticles2HepMCConverter module to avoid such overlap, but in this case the input label of RivetAnalyzer have to be changed. It's just a different way to workaround the problem.

Comment (davidlange6) [inline GeneratorInterface/RivetInterface/python/genParticles2HepMC_cfi.py]: the trouble with this solution is that you assume a particular process name that can change (and cause problems if ever this is run together with the generator.. Please change generator to something else. The usual convention is to follow your plugin name

Comment (Dr15Jones) [inline GeneratorInterface/RivetInterface/python/genParticles2HepMC_cfi.py]: You can tell an InputTag to skip the present process name when looking up data.

**Comment by jhgoh** [comment]
Module name is changed to follow usual name convention, generator -> genParticles2HepMC.
Instructions will be modified to use different inputTag after merging the PR.
Same changes applied to the backport PR #10321 .

**Comment by jhgoh** [comment]
Rebased to recent IB and fixed conflicts.

**Comment by davidlange6** [comment]
@cmsbuild, please test

**Comment by jhgoh** [comment]
Test is not started in some reason. @davidlange6, could you re-trigger the test?

**Comment by jhgoh** [comment]
Can this technical changes be merged?

**Comment by xjanssen** [comment]
Hi,
What is the status of this PR. I was checking 7_6_0_pre7 before starting installing again a new rivet version and it seems the rivet interface is still crashing due to this PR not being merged ...

**Comment by davidlange6** [comment]
it needs a generators signature. not sure why it is still outstanding.

**Comment by covarell** [comment]
+1
